typescript/trans/desugaring/desugar.str

```diff
@@ -50,7 +50,8 @@ rules
   
   desugar: ModuleItemsWithExport(items, item, items3) -> <concat> [items, [item], items3]
   desugar: ModuleItemsWithImport(items, item, items3) -> <concat> [items, [item], items3]
-	
+  
+  desugar: MethodFunction(name, callSignature, body) -> Function(FunctionName(name), callSignature, body)
```

Comment:
This desugaring is incorrect as long as we don't have functions as values. This is because during `Function` type checking, the name of the function is declared in the `Value` namespace, but for classes this should be in the `Field` namespace.

I suggest we change the current `Function` to `FunctionDeclaration` and add a AST constructor for function *values*. Then we can simply desugar `MethodFunction` to a `ClassProperty` with `Function` as value.